### ISSUE TITLE
Can't Create Multiple Partitions.

### ISSUE THREAD

**Issue by karcaw** [retr0h/cookbook-parted#38]
Using this cookbook i cant create multiple partitions of the same type.  I want to split a device into multiple partitions, but after the first one all the rest get skipped.

**Comment by retr0h** [comment]
@karcaw show me some examples.

**Comment by karcaw** [comment]
parted_disk "#{disks}" do
    label_type "gpt"
    action :mklabel
end

for i in 0..16 do
  s = i*20480
  e = s+20480
  parted_disk "#{disks}" do
    part_type   "primary"
    part_start  "#{s}"
    part_end    "#{e}"
    action :mkpart
  end
end

**Comment by retr0h** [comment]
@karcaw Ah you're suffering from a problem with chef.  Your resources can't have the same name.  Try:

`parted_disk "#{disks} for #{i}" do`

**Comment by karcaw** [comment]
This doesn't work because the not_if triggers only on the type of partition

new code: 
```
for i in 0..17 do
  s = i*20480
  e = s+20480
  parted_disk "#{disks} for #{i}" do
    device      "#{disks}"
    part_type   "primary"
    part_start  "#{s}"
    part_end    "#{e}"
    action :mkpart
  end
end
```

I had to add a device line because the for # gets put on the command line.

**Comment by Azrael808** [comment]
I'm having this problem myself... As I only need a secondary partition (and not a larger amount), I tried creating a `logical` partition inside an `extended` as a workaround, but that failed too because `logical` appears in the `print` output from `parted` no matter what (I've submitted a PR to fix that issue - #41  ).

FWIW, I think that the guard test could be improved, but it might require the creation of additional LWRP, one for each partition type... Just thinking out loud here. More than happy to have a look at implementing that if I have the time. :smile:
